controllers/floor_intake.py

```diff
@@ -25,16 +26,36 @@ def intake(self) -> None:
     @state(first=True, must_finish=True)
     def intaking(self, initial_call: bool):
         if self.algae_manipulator_component.has_algae():
-            self.next_state("feeling")
+            self.next_state("retracting")
             return
 
         self.intake_component.intake()
 
+        if (
+            self.intake_component.deploy_motor_current() > 100
```

Comment:
The state machine shouldn't care about the current going through the motor. Can you describe what this check is a proxy for, and make that a method on the component?